test/validations/greater-than.spec.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * @adonisjs/validator
+ *
+ * (c) Harminder Virk <virk@adonisjs.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+import test from 'japa'
+
+import { rules } from '../../src/Rules'
+import { schema } from '../../src/Schema'
+import { validate } from '../fixtures/rules/index'
+import { MessagesBag } from '../../src/MessagesBag'
+import { ApiErrorReporter } from '../../src/ErrorReporter'
+import { greaterThan } from '../../src/Validations/miscellaneous/greaterThan'
+import { NodeSubType } from '@ioc:Adonis/Core/Validator'
+
+function compile(greaterThanValue: any, subtype?: NodeSubType) {
+  // Regex Example for tax id validation from Brazil
+  return greaterThan.compile(
+    'literal',
+    subtype || 'string',
+    rules.greaterThan(greaterThanValue).options,
+    {}
+  )
+}
+
+test.group('greaterThan', () => {
+  validate(greaterThan, test, 20, 18, compile(18))
+
+  test('report error when value fails the greaterThan validation', (assert) => {
+    const reporter = new ApiErrorReporter(new MessagesBag({}), false)
+    greaterThan.validate(2, compile(1).compiledOptions, {
```

Comment:
In this case `2 > 1`. So shouldn't the validation pass?

Reply:
ye bhi samaj me nahi aya.